### PR TITLE
Clarifying DistributedContext purpose

### PR DESCRIPTION
**Pull request by toumorokoshi**
The DistributedContext section in the overview did not include examples,
and started by describing the data structure rather than the purpose.

Describing the purpose and use cases first will help better understand
the use cases.

### CODE REVIEW

File: specification/overview.md [hunk 1]
```diff
@@ -183,19 +183,18 @@ document.
 
 ## DistributedContext
 
-**DistributedContext** is an abstract data type that represents collection of entries.
-Each key of **DistributedContext** is associated with exactly one value. **DistributedContext** is serializable,
-to facilitate propagating it not only inside the process but also across process boundaries.
+The **DistributedContext** exists to store labels that describe the context of an operation an application performs. It is intended to enable context that are custom to the application or integrations in contrast to other contexts, such as `SpanContext`. For a specific operation, there should not be multiple **DistributedContext** instances.
```
Comment (Oberon00) [inline specification/overview.md]:
```suggestion
The **DistributedContext** exists to store labels that describe the context of an operation an application performs. It is intended to enable context that are custom to the application or integrations in contrast to other contexts, such as `SpanContext`. Only one **DistributedContext** should be associated with any particular operation.
```

Also, why is DistributedContext in bold font?

Comment (Oberon00) [inline specification/overview.md]:
```suggestion
The **DistributedContext** stores labels that describe the context of an operation an application performs. It is intended to enable contexts that are custom to the application or integrations in contrast to other contexts, such as `SpanContext`. For a specific operation, there should not be multiple **DistributedContext** instances.
```

Comment (Oberon00) [inline specification/overview.md]: > It is intended to enable contexts that are custom to the application or integrations

Is that so? I thought the Distributed Context, as specified now, was more or less what TagMap was in OpenCensus. But Ted commented in #242 that there will be a new proposal this week.

File: specification/overview.md [hunk 2]
```diff
@@ -205,7 +204,7 @@ a length greater than zero and less than 256.
 - **EntryMetadata** contains properties associated with an **Entry**.
 For now only the property **EntryTTL** is defined.
 - **EntryTTL** is an integer that represents number of hops an entry can propagate.
-Anytime a sender serializes an entry, sends it over the wire and receiver unserializes
+Anytime a sender serializes an entry, sends it over the wire and receiver deserializes
```
Comment (Oberon00) [inline specification/overview.md]:
```suggestion
Anytime a sender serializes an entry, sends it over the wire and a receiver deserializes
```

File: specification/overview.md (hunk 1 above)
Comment (toumorokoshi) [reply specification/overview.md]: I'm not sure. It's the convention in the previous version. Is there a convention on style we can refer to?

Comment (toumorokoshi) [reply specification/overview.md]: apparently github only lets me add one suggestion per line to a batch. I'll add this in a second commit.

Comment (toumorokoshi) [reply specification/overview.md]: I'm struggling for the right way to word this, but I wanted a way to help someone understand that this is a more free-form set of tags / labels that developers can use for their own stack's purposes.

For example, at Zillow we will probably use this to propagate originating service, which there is no spec around. 

When I say "integrations" I was thinking about telemetry SaaS providers. I'm sure there's keys that are valuable to propagate for them (maybe also and originating service key). 

Of course I may have misunderstood this. And yes maybe Ted will have a completely new proposal that changes the purpose dramatically.

File: specification/overview.md [hunk 3]
```diff
@@ -183,19 +183,18 @@ document.
...
-to facilitate propagating it not only inside the process but also across process boundaries.
+The **DistributedContext** exists to store labels that describe the context of an operation an application performs. It is intended to enable context that are custom to the application or integrations in contrast to other contexts, such as `SpanContext`. Only one **DistributedContext** should be associated with any particular operation.
+
+For example, a web service can benefit from including context around what service has sent the request. Or a SaaS provider can include context about the API user or token that is responsible for that request. These values can be consumed from **DistributedContext** and used as an additional dimension for a metric, or additional context for logs and traces.
 
-**DistributedContext** is used to annotate telemetry with the name:value pair **Entry**.
-Those values can be used to add dimension to the metric or additional contest properties to logs and traces.
+**DistributedContext** is a collection of key-value `Entry` pairs, with each key of associated with exactly one value. **DistributedContext** is serializable,
```
Comment (c24t) [inline specification/overview.md]:
```suggestion
**DistributedContext** is a collection of key-value `Entry` pairs, with each key associated with exactly one value. **DistributedContext** is serializable,
```

File: specification/overview.md [hunk 4]
```diff
@@ -183,19 +183,18 @@ document.
 
 ## DistributedContext
 
-**DistributedContext** is an abstract data type that represents collection of entries.
-Each key of **DistributedContext** is associated with exactly one value. **DistributedContext** is serializable,
-to facilitate propagating it not only inside the process but also across process boundaries.
+The **DistributedContext** exists to store labels that describe the context of an operation an application performs. It is intended to enable context that are custom to the application or integrations in contrast to other contexts, such as `SpanContext`. Only one **DistributedContext** should be associated with any particular operation.
```
Comment (c24t) [inline specification/overview.md]:
```suggestion
The **DistributedContext** exists to store labels that describe the context of an operation an application performs. It is intended to enable context that is custom to the application or integrations in contrast to other contexts, such as `SpanContext`. Only one **DistributedContext** should be associated with any particular operation.
```

Assuming you want "context" to be singular here, I don't know about "integrations" though.

**Comment by yurishkuro** [comment]
There are a couple proposals in flight (https://github.com/open-telemetry/oteps/pull/42, https://github.com/open-telemetry/oteps/pull/36), which will result in "distributed context" term going away, so maybe worth waiting before merging this PR.

**Comment by SergeyKanzhelev** [comment]
@yurishkuro this PR doesn't make anything worse in the current spec. Context separation is planned for v0.3 which is few weeks away so I'd suggest we will merge this PR for now.

**Pull request merged**